tests/test_TFEngine.py

```diff
@@ -536,6 +536,70 @@ def test_engine_rec_subnet_count():
   engine.finalize()
 
 
+def check_engine_end_layer(extra_rec_kwargs=None):
```

Comment:
Why do you need that as an extra function? There is only a single place where you call it.